metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -277,15 +336,17 @@ def get_datasource_urn_from_id(self, datasource_id):
 
         if database_id and table_name:
             return make_dataset_urn(
-                platform=platform,
+                platform=self.platform,
```

Comment:
this should be the external platform e.g. snowflake/bigquery

```suggestion
                platform=platform,
```